interfaces/functions.yaml

```diff
@@ -804,6 +804,9 @@ igraph_maxdegree:
 igraph_density:
     PARAMS: GRAPH graph, OUT REAL res, BOOLEAN loops=False
 
+igraph_mean_degree:
+    PARAMS: GRAPH graph, OUT REAL res, BOOLEAN loops=True
```

Comment:
Here's an inconsistency with the default value of `loops` for `igraph_density`. If we are heading towards consistent and predictable defaults, we should change one or the other.

Reply:
I know this _looks_ inconsistent (it grabbed my attention as well), but in reality it isn't. I think we should leave the default as they are now, for both functions.

These two `loops` parameters have very different meanings.

 - In `igraph_density()` it controls whether the graph is assumed to _potentially_ have self-loops, and has an impact on computing the largest possible number of edges. This function doesn't check if the graph _actually_ has self-loops.
 - In `igraph_mean_degree()` it controls whether self-loops are discarded if they happen to be present. `loops=false`, i.e. discarding loops, causes the runtime to be linear in the edge count. `loops=true` is constant time, and suitable for the majority of use cases.

In the typical practical use case, people work with a network that _does not_ have self-loops, and _cannot_ have self-loops. Thus in `igraph_density()`, `loops=false` is mandatory for the result to make sense. `igraph_mean_degree()` would give the same result regardless of whether `loops=true` of `loops=false` is used, but `loops=true` does a constant-time calculation, while `loops=false` is linear.

Reply:
Something I thought about was whether `igraph_density()` should issue a warning when we know from the cache that the graph does have loops, but `loops=false` was set. In this case the result is meaningless, so a warning makes sense. I did not add such a warning, conditional on what is cached, because users may take a lack of warning as assurance that there are no problems.